store/aerospike/aerospike.go

```diff
@@ -0,0 +1,322 @@
+	"time"
+
+	"github.com/abronan/valkeyrie"
+	"github.com/abronan/valkeyrie/store"
+	api "github.com/aerospike/aerospike-client-go"
+	"github.com/aerospike/aerospike-client-go/types"
+)
+
+// Aerospike is the receiver type for the
+// Store interface
+type Aerospike struct {
+	sync.Mutex
+	client    *api.Client
+	namespace string
+}
+
+var (
+	// ErrAerospikeNamespaceMissing is thrown when Namespace config option is missing
+	ErrAerospikeNamespaceMissing = errors.New("Namespace (Bucket) config option missing")
+)
+
+// Register registers aerospike to valkeyrie
+func Register() {
+	valkeyrie.AddStore(store.AEROSPIKE, New)
+}
+
+// New creates a new Aerospike client given a list
+// of endpoints and optional tls config
+func New(endpoints []string, options *store.Config) (store.Store, error) {
+
+	s := &Aerospike{}
+
+	hosts := []*api.Host{}
+	for _, endpoint := range endpoints {
+		h := strings.Split(endpoint, ":")
+		p, err := strconv.Atoi(h[1])
+		if err != nil {
+			return nil, err
+		}
+		hosts = append(hosts, api.NewHost(h[0], p))
+	}
+
+	// Set options
+	if (options == nil) || (len(options.Bucket) == 0) {
+		return nil, ErrAerospikeNamespaceMissing
+	}
+	s.namespace = options.Bucket
+	policy := api.NewClientPolicy()
+	if options != nil {
+		if options.TLS != nil {
+			policy.TlsConfig = options.TLS
+		}
+		if options.ConnectionTimeout != 0 {
+			policy.Timeout = options.ConnectionTimeout
+		}
+		if options.Username != "" {
+			policy.User = options.Username
+		}
+		if options.Password != "" {
+			policy.Password = options.Password
+		}
+		if options.SyncPeriod != 0 {
+			policy.TendInterval = options.SyncPeriod
+		}
+	}
+
+	// Creates a new client
+	client, err := api.NewClientWithPolicyAndHost(policy, hosts...)
+	if err != nil {
+		return nil, err
+	}
+
+	regTask, err := client.RegisterUDF(nil, []byte(listUDF), "valkeyrie.lua", api.LUA)
+	if err != nil {
+		return nil, err
+	}
+	err = <-regTask.OnComplete()
+	if err != nil {
+		return nil, err
+	}
+
+	s.client = client
+
+	return s, nil
+}
+
+// newKey initializes a key from namespace
+func (s *Aerospike) newKey(key string) (*api.Key, error) {
+	var akey *api.Key
+	ikey, err := strconv.ParseInt(key, 10, 64)
+	if err == nil {
+		akey, err = api.NewKey(s.namespace, "", ikey)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		akey, err = api.NewKey(s.namespace, "", key)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return akey, err
+}
+
+// Get the value at "key", returns the last modified index
+// to use in conjunction to CAS calls
+func (s *Aerospike) Get(key string, opts *store.ReadOptions) (*store.KVPair, error) {
+	policy := api.NewPolicy()
+	akey, err := s.newKey(key)
+	if err != nil {
+		return nil, err
+	}
+
+	record, err := s.client.Get(policy, akey)
+	if record == nil {
+		return nil, store.ErrKeyNotFound
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return &store.KVPair{Key: key, Value: record.Bins["bin"].([]byte), LastIndex: uint64(record.Generation)}, nil
```

Comment:
Shouldn't we try to check if the existing map entry exists before trying to cast it into a `[]byte`? If this is nil this would end up in a panic I think (but not sure)?

Reply:
did return the bins with any name